src/main/autoeq.ts

```diff
@@ -18,18 +19,75 @@ export const getAutoEqResponseList = (device: string) => {
 };
 
 export const getAutoEqPreset = (device: string, response: string) => {
-  // TODO: parse equalizer apo file and return data as a state
+  const PREAMP_REGEX = new RegExp('Preamp:[\\s]*([-][\\d]*[.][\\d]*)');
+
+  // filter #: PK/LS/HS <freq> HZ Gain <gain> dB Q <quality>
+  const LINE_REGEX = new RegExp(
+    'Filter ([\\d]+)[:][a-zA-Z\\s]+(PK|LS|HS)[a-zA-Z\\s]+([\\d]+)[a-zA-Z\\s]+([-]?[\\d]*[.][\\d]*)[a-zA-Z\\s]+([\\d]*[.][\\d]*)'
+  );
+
+  let preAmpParsed = 0;
+  const filterIds: string[] = [];
+  const filterTypes: FilterTypeEnum[] = [];
+  const frequencies: number[] = [];
+  const gains: number[] = [];
+  const qualities: number[] = [];
+
+  const file = fs.readFileSync(path.join(AUTOEQ_DIR, device, response), 'utf8');
+  file.split('\n').forEach((line) => {
+    let match = line.match(PREAMP_REGEX);
+    if (match) {
+      if (match.length !== 2) {
+        throw new Error(`Regex match error on ${line}`);
+      }
+      preAmpParsed = parseFloat(match[1]);
+      return;
+    }
+    match = line.match(LINE_REGEX);
+    if (match) {
+      if (match.length !== 6) {
+        throw new Error(`Regex match error on ${line}`);
+      }
+      const filterId = uid(8);
+      let filterType;
+      if (match[2] === 'PK') {
+        filterType = FilterTypeEnum.PK;
+      } else if (match[2] === 'LS') {
+        filterType = FilterTypeEnum.LSC;
+      } else if (match[2] === 'HS') {
+        filterType = FilterTypeEnum.HSC;
+      } else {
+        throw new Error('Unknown filter type');
```

Comment:
Is there a risk that this would cause issues with the AutoEQ presets? I.e. are we certain that none of the AutoEQ options use filter types that we haven't included in our app yet? Would it be safer to use a default filter type rather than throw an error...? Thoughts?

Reply:
Good point. I think we should use Peak filter as default. @h39s thoughts?